src/hooks/useDisputes.js

```diff
@@ -38,37 +40,99 @@ export default function useDisputes() {
+      disputes: disputes.map((dispute, i) => {
+        const [disputeDescription] = getDisputeInfoFromMetadata(
+          dispute.metadata
+        )
+        return {
+          ...dispute,
+          description: disputeDescription,
+          ...disputesPhases[i],
+        }
+      }),
     }
   }, [disputesPhases, disputes, disputesPhasesKey, error]) // eslint-disable-line react-hooks/exhaustive-deps
 }
 
 export function useDispute(disputeId) {
+  const [disputeProcessed, setDisputeProcessed] = useState()
   const courtConfig = useCourtConfig()
   const now = useNow() // TODO: use court clock
   const { dispute, fetching } = useSingleDisputeSubscription(disputeId)
+  useEffect(() => {
+    const fetchDataFromIpfs = async () => {
+      if (dispute) {
+        const [
+          disputeDescription,
+          disputeMetadata,
+        ] = getDisputeInfoFromMetadata(dispute.metadata)
+
+        if (!disputeMetadata) {
+          return setDisputeProcessed({ ...dispute, error: true })
+        }
+        const ipfsPath = disputeMetadata.replace(/^ipfs:/, '')
+
+        if (isIPFS.cidPath(ipfsPath)) {
+          const { data, error } = await ipfsGet(ipfsPath)
+          if (error) {
+            return setDisputeProcessed({ ...dispute, error: true })
+          }
+          try {
+            const parsedDisputeData = JSON.parse(data)
+            const agreementText = parsedDisputeData.agreementText.replace(
+              /^.\//,
+              ''
+            )
+            const agreementUrl = resolvePathname(
+              agreementText,
+              `${IPFS_ENDPOINT}/${ipfsPath}`
+            )
+            return setDisputeProcessed({
+              ...dispute,
+              description: parsedDisputeData.description || disputeDescription,
+              agreementText: agreementText || '',
+              agreementUrl,
+              defendant: parsedDisputeData.defendant || '',
+              plaintiff: parsedDisputeData.plaintiff || '',
+              error: false,
+            })
+          } catch (err) {
+            return setDisputeProcessed({ ...dispute, description: data })
+          }
+        }
+        return setDisputeProcessed({ ...dispute, error: true })
+      }
+    }
+
+    fetchDataFromIpfs()
+  }, [dispute])
 
   const disputePhase = getPhaseAndTransition(dispute, courtConfig, now)
   const disputePhaseKey = disputePhase
     ? convertToString(Object.values(disputePhase)[0])
     : ''
 
   return useMemo(() => {
-    if (fetching) {
-      return { fetching }
+    if (fetching || !disputeProcessed) {
+      return { fetching: true }
     }
 
     return {
       dispute: {
-        ...dispute,
-        ...processDisputeData(dispute),
+        ...disputeProcessed,
         ...disputePhase,
       },
       fetching,
     }
-  }, [dispute, disputePhaseKey]) // eslint-disable-line react-hooks/exhaustive-deps
+  }, [disputeProcessed, dispute, disputePhaseKey]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Still wondering on this; which dependency are we not interested in having here? And why?